PluginCore/PluginCore/Helpers/PlatformHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Win32;
+using System;
+
+namespace PluginCore.Helpers
+{
+    public static class PlatformHelper
+    {
+        // INIT
+        private static Boolean isRunningOnWindows = (Environment.OSVersion.Platform != PlatformID.Unix) && (Environment.OSVersion.Platform != PlatformID.MacOSX);
+        private static Boolean isRunningOnMono = Type.GetType("Mono.Runtime") != null;
+
+        /// <summary>
+        /// Checks if we are running on Windows
+        /// </summary>
+        public static Boolean IsRunningOnWindows()
+        {
+            return isRunningOnWindows;
+        }
+
+        /// <summary>
+        ///  Checks if we are running on Wine
+        /// </summary>
+        public static Boolean isRunningOnWine()
+        {
+            return isRunningOnWindows && (Registry.LocalMachine.OpenSubKey(@"Software\Wine\") != null);
+        }
+
+        /// <summary>
+        /// Checks if we are running on Mono
+        /// </summary>
+        public static Boolean IsRunningOnMono()
+        {
+            return isRunningOnMono;
+        }
```

Comment:
It's just a minor thing about coding style, but
I think it's better to have them as getter properties than methods, since (unlike what the xml doc says) the methods only returns a fixed stored values instead of doing the checks on each call.
Usually a method is used instead of a getter property in cases where an extensive calculation is done on each call.